### PR TITLE
feat(dedicated.license): enable license upgrade with new routes

### PR DESCRIPTION
**Pull request by sachinrameshn**
| Question         | Answer
| ---------------- | ---
| Branch?          | `master`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #MANAGER-14654
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~~Breaking change is mentioned in relevant commits~~

## Description

Need to enable upgrade for all below licenses with the help of new apis
- cPanel
- Windows
- SQL server
- Plesk

### CODE REVIEW

**Review by qpavy — DISMISSED** [review]

**Review by mhelhali-soufien — APPROVED** [review]

**Review by qpavy — DISMISSED** [review]

File: packages/manager/apps/dedicated/client/app/license/license.service.js [hunk 1]
```diff
@@ -62,6 +62,94 @@ export default /* @ngInject */ function LicenseService(
...
+  this.getActualUpgradeVersion = function getActualUpgradeVersion(serviceId) {
+    return OvhHttp.get(`/services/{serviceId}`, {
+      rootPath: 'apiv6',
+      urlParams: {
+        serviceId,
+      },
+    });
+  };
+
+  this.getAllUpgradeVersion = function getAllUpgradeVersion(serviceId) {
```
Comment (mhelhali-soufien) [inline packages/manager/apps/dedicated/client/app/license/license.service.js:74]: same

File: packages/manager/apps/dedicated/client/app/license/license.service.js [hunk 2]
```diff
@@ -62,6 +62,94 @@ export default /* @ngInject */ function LicenseService(
...
+    return OvhHttp.get(`/services/{serviceId}/upgrade`, {
+      rootPath: 'apiv6',
+      urlParams: {
+        serviceId,
+      },
+    });
+  };
+
+  this.getUpgradeVersion = function getUpgradeVersion(serviceId) {
```
Comment (mhelhali-soufien) [inline packages/manager/apps/dedicated/client/app/license/license.service.js:83]: same for all

File: packages/manager/apps/dedicated/client/app/license/license.service.js [hunk 3]
```diff
@@ -62,6 +62,94 @@ export default /* @ngInject */ function LicenseService(
     }).then((data) => data.options);
   };
 
+  this.getActualUpgradeVersion = function getActualUpgradeVersion(serviceId) {
```
[the remark this answers is not in the document]
Comment (sachinrameshn) [reply packages/manager/apps/dedicated/client/app/license/license.service.js]: Applied as suggested.

File: packages/manager/apps/dedicated/client/app/license/license.service.js [hunk 4]
```diff
@@ -62,6 +62,103 @@ export default /* @ngInject */ function LicenseService(
...
+            return result.upgradableVersions;
+        }
+      });
+  };
+
+  this.getFilteredUpgradeVersions = function getFilteredUpgradeVersions(
+    results,
+    expression,
+    planCode,
+  ) {
```
Comment (qpavy) [inline packages/manager/apps/dedicated/client/app/license/license.service.js:130]:
```suggestion
  this.getFilteredUpgradeVersions = (
    results,
    expression,
    planCode,
  ) => {
```

File: packages/manager/apps/dedicated/client/app/license/license.service.js [hunk 5]
```diff
@@ -62,6 +62,103 @@ export default /* @ngInject */ function LicenseService(
...
+      return results.filter((version) => {
+        return version.planCode.endsWith(`${expression}-ca`);
+      });
+    }
+    return results.filter((version) => {
+      return version.planCode.includes(expression);
+    });
+  };
+
+  this.simulateLicenseUpgrade = function simulateLicenseUpgrade(data) {
```
Comment (qpavy) [inline packages/manager/apps/dedicated/client/app/license/license.service.js:141]:
```suggestion
  this.simulateLicenseUpgrade = (data) => {
```

File: packages/manager/apps/dedicated/client/app/license/license.service.js [hunk 6]
```diff
@@ -62,6 +62,103 @@ export default /* @ngInject */ function LicenseService(
...
+      rootPath: 'apiv6',
+      urlParams: {
+        serviceName: data.serviceId,
+        planCode: data.options.version.planCode,
+      },
+      data: payload,
+    });
+  };
+
+  this.getServiceUpgradeParams = function getServiceUpgradeParams() {
```
Comment (qpavy) [inline packages/manager/apps/dedicated/client/app/license/license.service.js:153]:
```suggestion
  this.getServiceUpgradeParams = () => {
```

**Review by qpavy — DISMISSED** [review]

**Review by mhelhali-soufien — APPROVED** [review]

**Review by qpavy — APPROVED** [review]

**Review by tristanwagner — APPROVED** [review]

**Pull request merged**